transport/client.go

```diff
@@ -311,7 +315,7 @@ func Dial(address string, tr *Transport) (*tls.Conn, error) {
 
 	for _, chain := range state.VerifiedChains {
 		for _, cert := range chain {
-			revoked, ok := revoke.VerifyCertificate(cert)
```

Comment:
I feel there are several issues we are trying to solve here. I want to clarify with you together to get our PR defined by a feature request. 
Let me digress a little bit first.
1. self-signed certs probably never needs revocation or CRL, since it is easier to just throw those away from trust stores.
2. With file based CRL, we may simplify the CRL distribution (we don't need a server anyway), this will help a customized internal PKI.
3. Public CA probably don't need this file-based CRL feature.

We want to do 2 as an option right? and we don't want to complicate 3. 
Now back to this change. What I saw here at this line is that tr.RevokeChecker is not different from a defaultChecker which can be defined universally. Maybe we will want a customized checker later, but not now, at least from this change I can't justify it.